compose/foundation/foundation/src/skikoMain/kotlin/androidx/compose/foundation/text/CupertionTextFieldPointerModifier.skiko.kt

```diff
@@ -132,24 +133,86 @@ private fun getTapHandlerModifier(
     }
 }
 
+/**
+ * Returns a modifier which allows to precisely move the caret in the text by drag gesture after long press
+ *
+ * @param state The state of the text field.
+ * @param offsetMapping The offset mapping of the text field.
+ * @return A modifier that handles long press and drag gestures.
+ */
 @Composable
-private fun getSelectionModifier(manager: TextFieldSelectionManager): Modifier {
-    val currentManager by rememberUpdatedState(manager)
+private fun getLongPressHandlerModifier(
+    state: TextFieldState,
+    offsetMapping: OffsetMapping
+): Modifier {
+    val currentState by rememberUpdatedState(state)
+    val currentOffsetMapping by rememberUpdatedState(offsetMapping)
+    var dragTotalDistance = remember { Offset.Zero }
+    var dragBeginOffset = remember { Offset.Zero }
+    var currentDragPosition: Offset?
+
+    val longTapActionsObserver = object : TextDragObserver {
```

Comment:
As an alternative to `remember`, we can move it to the coroutine itself:
```
return Modifier.pointerInput(Unit) {
    val longTapActionsObserver = object : TextDragObserver {
        ...
    }
    detectDragGesturesAfterLongPress(
```
This will look better

Reply:
> similar effect like remember

Yes, the coroutine is only created once, and suspends on `detectDragGesturesAfterLongPress`

> but more lazy?

It is created at another time, yes, but this only give us a better code (related pieces defined in one place.)